src/pages/AllProducts.js

```diff
@@ -0,0 +1,54 @@
+import React, { useState, useEffect } from "react";
+import Product from "../components/Product";
+import Sidebar from "../components/Sidebar";
+import Paging from "../components/Paging";
+import { useLocation } from "react-router-dom";
+//Hooks
+import { useFilteredProductsFromApi } from "../utils/hooks/useFilteredProductsFromApi";
+import { useCategoriesFromApi } from "../utils/hooks/useCategoriesFromApi";
+
+function useQuery() {
+  const { search } = useLocation();
+  return React.useMemo(() => new URLSearchParams(search), [search]);
+}
+
+const AllProducts = () => {
+  let query = useQuery();
+  
+  const { categories, categoriesAreLoading } = useCategoriesFromApi();
+  const { filteredProductsData, filteredProductsAreLoading } =
+    useFilteredProductsFromApi(query.get("category"), query.get("page"));
+  const [sidebarCategories, setItems] = useState(categories);
+  const [products, setProducts] = useState(filteredProductsData.results);
+
+  useEffect(() => {
+    setItems(categories);
+  }, [categories]);
+
+  useEffect(() => {
+    setProducts(filteredProductsData.results);
+    }, [filteredProductsData]);
+
+  if (categoriesAreLoading || filteredProductsAreLoading) {
+    return <h1>Loading...</h1>;
+  }
+
+  return (
+    <div>
+      <Sidebar categories={sidebarCategories} query={query.get("category")} />
+      {products && products.map((product) => {
+        const category = sidebarCategories.find(
+          (category) => category.id === product.categoryId
+        ).name;
+        return (
+          <div key={product.id}>
+            <Product product={product} category={category} />
+          </div>
+        );
+      })}
+      {filteredProductsData &&<Paging page={filteredProductsData.page} pages={filteredProductsData.total_pages} query={query}/>}
+    </div>
```

Comment:
```suggestion
    <div>
      <Sidebar categories={sidebarCategories} query={query.get("category")} />
      <div style = {{display:"grid", gridTemplateColumns:"1fr 1fr 1fr 1fr", gridGap:"1rem"}}>
        {products &&
          products.map((product) => {
            const category = sidebarCategories.find(
              (category) => category.id === product.categoryId
            ).name;
            return (

                <Product key = {product.key} product={product} category={category} />

            );
          })}
      </div>
      {filteredProductsData && (
        <Paging
          page={filteredProductsData.page}
          pages={filteredProductsData.total_pages}
          query={query}
        />
      )}
    </div>
```

* Adding `style={{
          display: "grid",
          gridTemplateColumns: "1fr 1fr 1fr 1fr",
          gridGap: "1rem",
        }}`
        
 to the products container will make it look more like a grid, check out this [article](https://css-tricks.com/snippets/css/complete-guide-grid/) in css tricks where you can find more info about css grid

* The `<Product/>` component does not need a `<div>` wrapper